src/lib/krb5/ccache/ccfns.c

```diff
@@ -100,9 +101,14 @@ krb5_cc_store_cred(krb5_context context, krb5_ccache cache,
     if (!krb5_principal_compare(context, s1, s2)) {
         creds->server = s2;
         TRACE_CC_STORE_TKT(context, cache, creds);
-        /* remove any dups */
-        krb5_cc_remove_cred(context, cache, KRB5_TC_MATCH_AUTHDATA, creds);
-        ret = cache->ops->store(context, cache, creds);
+        ret = krb5_cc_retrieve_cred(context, cache, KRB5_TC_MATCH_AUTHDATA,
+                                    creds, &old_creds);
+        if (ret != KRB5_CC_NOTFOUND)
+            return ret;
+        if (ret == 0)
+            krb5_free_cred_contents(context, &old_creds);
```

Comment:
This line is unreachable.

Reply:
Which line? I don't see, please elaborate.

Reply:
If `ret == 0`, it would also be the case that `ret != KRB5_CC_NOTFOUND` (line 106) and be returned (line 107) before getting to the check (line 108).  This makes line 109 unreachable.

Reply:
Ah sorry, I see now.. I fixed it as it was meant to be.